### PR TITLE
feat: implement oif adapter

### LINKED ISSUE #18: Implement OIF Adapter
- **Description:** Implement adapter for OIF protocol according to the OIF GetQuote and SubmitIntent API standards.
- **Acceptance Criteria:**
    - Translates OIF-compatible requests into internal solver API calls
    - Converts internal solver responses back into OIF-compatible responses
    - Unit tests verifying protocol translation and field mapping
    - e2e test
    - edge case testing

### PR DESCRIPTION
**Pull request by zeljkoX**
# Summary

This PR will:
- implement OIF adapter
- add client caching for adapters to keep active connections
- add support for background job processing
- load assets on service start and refresh assets support for solver every 20min
- health check on service start and run it every 5 min
- improved config supportedAssets, they are used as global override. In most cases they do not need to be set as system will fetch supported tokens on service start and every 20min if config field is not defined.

Note: We can say that this is initial OIF Adapter implementation as OIF spec is not yet finalised and changes are expected.

Observations on OIF solver:
- camelCase is not used for all fields (partial PR created)
- interop addresses are not used in responses but they are used as part of requests (in discussion, probably will be applied on solver side)
- create order endpoint does not return fill order body (in discussion. Currently aggregator tries to fetch order after creating it to get full body)

Tested with solver demo in localhost, all endpoints working.

Depends on https://github.com/openintentsframework/oif-solver/pull/103

Closes #18 

## Checklist

- [x] Add a reference to related issues in the PR description.
- [x] Add unit tests if applicable.

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
       "adapter_id": "oif-v1",
-      "endpoint": "https://api.example.com/v1",
+      "endpoint": "http://127.0.0.1:3000/api",
       "timeout_ms": 1500,
       "enabled": true,
       "max_retries": 1,
       "headers": null,
-      "name": "Example Solver",
-      "description": "Example Solver Description",
-      "supported_assets": [
```
Comment (zeljkoX) [inline config/config.json:17]: supported_assets fields can be defined but it acts as an override for values. 

By default when not set we will automatically fetch supported assets from solver.

File: crates/types/src/models/interop_address.rs [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
 //! An ERC-7930 interoperable address is a binary-encoded string serialized as hex:
 //! ```
-//! 0x00010000010114D8DA6BF26964AF9D7EED9E03E53415D37AA96045
-//! ^^^^---------------------------------- Version: 4 bytes (decimal 1)
-//!     ^^^^------------------------------ ChainType: 2 bytes (CAIP namespace, e.g., eip155)
-//!         ^^---------------------------- ChainReferenceLength: 1 byte (decimal length)
-//!           ^^-------------------------- ChainReference: variable length (e.g., uint8 chain ID)
-//!             ^^------------------------ AddressLength: 1 byte (decimal length)
-//!               ^^^^^^^^^^^^^^^^^^^^^^^^ Address: variable length (e.g., 20 bytes for Ethereum)
+//! 0x010000011401D8DA6BF26964AF9D7EED9E03E53415D37AA96045
```
Comment (zeljkoX) [inline crates/types/src/models/interop_address.rs:11]: improvements needed in order to parse solver responses

File: crates/adapters/src/client_cache.rs [hunk 3]
```diff
@@ -0,0 +1,386 @@
+//! HTTP client cache for optimized connection management
+//!
+//! Provides per-solver client instances with connection pooling and keep-alive optimization.
+
+use dashmap::DashMap;
```
Comment (zeljkoX) [inline crates/adapters/src/client_cache.rs:5]: Client cache optimisations to keep alive connects between requests.
Used within adapters

File: crates/service/src/jobs/processor.rs [hunk 4]
```diff
@@ -0,0 +1,1237 @@
+//! Background job processor implementation
```
Comment (zeljkoX) [inline crates/service/src/jobs/processor.rs:1]: Introduced background job logic in order to fetch supported assets and do healthcheck on startup and on specific intervals.

Later this functionality would be used to:
- process metrics
- check order status
- ...

File: crates/types/src/adapters/models.rs [hunk 5]
```diff
@@ -236,19 +237,31 @@ pub struct GetOrderResponse {
 	pub order: OrderResponse,
 }
 
+/// Response containing order details.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "openapi", derive(ToSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct SubmitOrderResponse {
```
Comment (zeljkoX) [inline crates/types/src/adapters/models.rs:244]: If create order endpoint starts to return full order body this type could be deleted

File: config/config.json [hunk 6]
```diff
@@ -40,7 +31,7 @@
     }
   },
   "logging": {
-    "level": "info",
+    "level": "debug",
```
Comment (shahnami) [inline config/config.json:34]: nit; not sure if you want to revert this back to `info`?

File: crates/types/src/adapters/models.rs [hunk 7]
```diff
@@ -197,7 +198,7 @@ pub enum OrderStatus {
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct AssetAmount {
 	/// Asset address in ERC-7930 interoperable format
-	pub asset: InteropAddress,
+	pub asset: String,
```
[the remark this answers is not in the document]
Comment (shahnami) [reply crates/types/src/adapters/models.rs:201]: This is now updated in Solver as well.

File: config/config.json (hunk 6 above)
Comment (zeljkoX) [reply config/config.json:34]: Thanks for noticing. I will tackle improvements in upcoming PRs. This whole file will be renamed to config.example.json
